thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AnomaliesResource.java

```diff
@@ -665,11 +665,19 @@ private AnomalyDetails getAnomalyDetails(MergedAnomalyResultDTO mergedAnomaly, D
       }
 
       List<MergedAnomalyResultDTO> knownAnomalies = adInputContext.getKnownMergedAnomalies().get(dimensions);
-      // Known anomalies are ignored (the null parameter) because 1. we can reduce users' waiting time and 2. presentation
-      // data does not need to be as accurate as the one used for detecting anomalies
-      AnomalyTimelinesView anomalyTimelinesView =
-          anomalyFunction.getTimeSeriesView(metricTimeSeries, bucketMillis, anomalyFunctionSpec.getTopicMetric(),
-              anomalyWindowStart, anomalyWindowEnd, knownAnomalies);
+
+      AnomalyTimelinesView anomalyTimelinesView = null;
+      Map<String, String> anomalyProps = mergedAnomaly.getProperties();
+      // check if there is AnomalyTimelinesView in the Properties. If yes, use the AnomalyTimelinesView
+      if(anomalyProps.containsKey("anomalyTimelinesView")) {
+        anomalyTimelinesView = AnomalyTimelinesView.fromJsonString(
+            anomalyProps.get("anomalyTimelinesView"));
+      } else {
+        // Known anomalies are ignored (the null parameter) because 1. we can reduce users' waiting time and 2. presentation
```

Comment:
Please optimize the code here: If the anomaly provides it's own time series, then we don't need to retrieve time series either. You might want to move lines 653-667 into this block.